backend-config/types.go

```diff
@@ -57,6 +57,7 @@ type DestinationT struct {
 	RevisionID            string
```

Comment:
Here we are relying on the json library case-insensitive unmarshalling of `revisionId` & `id`, something that is about to [change in json v2](https://github.com/golang/go/discussions/63397#:~:text=Case%2Dinsensitive%20unmarshaling,Go%20map%20lookup.).
Not only that, when we marshal the destination struct ourselves to pass it to the transformer we are using a different field name, `RevisionID` & `ID`.
Being it this situation already, the risk of causing a regression by fixing this is high :(

Reply:
what do you suggest to do here?

Reply:
we can't update this now very easily as we have config backend -> rudder server, rudder server -> transformer different contract this, we need dedicated effort this.

Reply:
I know, here I am merely highlighting some sins of the past that are going to trouble us in the future